tee-worker/omni-executor/intent/executors/cross-chain/src/cross_chain.rs

```diff
@@ -164,19 +165,34 @@ impl<
 						}),
 					))
 				} else {
-					// TODO: this should be abstracted away
-					let (payout_amount, payout_amount_u256) = self
-						.do_binance_swap_sol_to_bsc(
+					let chain_transfer_client = Arc::new(SolanaTransferClient::new(
+						self.solana_client.clone(),
+						RemoteSolanaSigner::new(
+							self.pumpx_signer_client.clone(),
+							pumpx_config.wallet_index,
 							omni_account,
+							Handle::current(),
+						),
+					));
+					let (payout_amount, payout_amount_u256) = self
+						.do_binance_swap(
+							&self.evm_accounting_contract_client,
+							chain_transfer_client,
 							swap_order.from_asset.clone(),
 							pumpx_config.from_amount.clone(),
 							from_address,
-							pumpx_config.wallet_index,
+							BinanceNetwork::Bsc,
+							BinanceCoin::Bnb,
 							false,
 						)
 						.await?;
 
-					self.do_payout_sol_to_bsc(&payout_address, &payout_amount_u256).await?;
+					do_payout(
```

Comment:
Just curious: why `do_binance_swap` is a member fn while `do_payout` isn't? Shall we make them consistent?